### PR TITLE
Video: various updates

### PR DESCRIPTION
**Pull request by jerpelea**
## Summary
various improvements for the video implementation

## Impact
video implementation

## Testing
Spresense

### CODE REVIEW

File: drivers/video/isx012.c [hunk 1]
```diff
@@ -644,26 +663,33 @@ static uint16_t isx012_getreg(isx012_dev_t *priv,
...
-  /* Restart and read 16bits from the register */
+  i2c_unlock();
 
-  ret = i2c_read(priv->i2c, &config, (uint8_t *)buffer, regsize);
-  if (ret < 0)
+  if (ret >= 0)
     {
-      verr("i2c_read failed: %d\n", ret);
-      return 0;
+      memcpy((uint8_t *)&regval, (uint8_t *)buffer, regsize);
```
Comment (pkarashchenko) [inline drivers/video/isx012.c:690]:
```suggestion
      memcpy((FAR uint8_t *)&regval, (FAR uint8_t *)buffer, regsize);
```

File: drivers/video/isx012.c [hunk 2]
```diff
@@ -1236,6 +1241,45 @@ int init_isx012(FAR struct isx012_dev_s *priv)
   return ret;
 }
 
+static void get_chipid(uint32_t *l, uint32_t *h)
```
Comment (xiaoxiang781216) [inline drivers/video/isx012.c]: add FAR

File: drivers/video/isx012.c [hunk 3]
```diff
@@ -215,8 +220,10 @@ static bool is_movie_needed(uint8_t fmt, uint8_t fps);
 
 /* image sensor device operations interface */
 
+static bool isx012_is_available(void);
 static int isx012_init(void);
 static int isx012_uninit(void);
+static const char *isx012_get_driver_name(void);
```
[the remark this answers is not in the document]
Comment (pkarashchenko) [reply drivers/video/isx012.c:226]: I think here the change also should be applied

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply drivers/video/isx012.c:226]: This one not fix @jerpelea ?

File: drivers/video/isx012.c [hunk 4]
```diff
@@ -687,6 +713,8 @@ static int isx012_putreg(isx012_dev_t *priv,
 
   memcpy((uint8_t *)&buffer[2], (uint8_t *)&regval, regsize);
```
Comment (pkarashchenko) [inline drivers/video/isx012.c:714]:
```suggestion
  memcpy((FAR uint8_t *)&buffer[2], (FAR uint8_t *)&regval, regsize);
```

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

File: drivers/video/isx012.c [hunk 5]
```diff
@@ -2870,24 +2947,26 @@ static int isx012_set_shd(FAR isx012_dev_t *priv)
...
+    {
+      verr("Failed to register ops to video driver.\n");
+      return ret;
+    }
 
   /* Initialize other information */
 
   priv->state      = STATE_ISX012_POWEROFF;
 
+  nxsem_init(&priv->i2c_lock, 0, 1);
```
Comment (pkarashchenko) [inline drivers/video/isx012.c:2968]: We can try to start using `nxmutex` in some further PRs.

Comment (jerpelea) [reply drivers/video/isx012.c:2968]: ok

**Pull request merged**